common/src/IO/WorldReader.cpp

```diff
@@ -43,6 +44,31 @@ namespace TrenchBroom {
             m_world->disableNodeTreeUpdates();
         }
 
+        std::unique_ptr<Model::WorldNode> WorldReader::read(std::string_view str, const std::vector<Model::MapFormat>& mapFormatsToTry, const vm::bbox3& worldBounds, ParserStatus& status) {
+            std::optional<ParserException> lastException;
+
+            for (const auto mapFormat : mapFormatsToTry) {
+                if (mapFormat == Model::MapFormat::Unknown) {
+                    continue;
+                }
+
+                try {
+                    WorldReader reader(str, mapFormat);
+                    return reader.read(worldBounds, status);
+                } catch (const ParserException& e) {
+                    lastException = e;
+                }
+            }
+
+            if (lastException) {
+                // No format parsed successfully. Just throw the parse error from the last one.
+                throw *lastException;
```

Comment:
Just throwing the last exception that happened here takes away useful information that a user needs to fix errors in their map files. I think you need to capture all of them and display them in a useful way, like a table of the formats that were tried with the corresponding error messages that occurred.

Reply:
Sounds good, will try to add this

Reply:
Please also consider my other comment that I left after the review. If you want, we can discuss it further before proceeding!